python/examples/sklearn_vw.ipynb

```diff
@@ -44,50 +39,31 @@
   },
   {
    "cell_type": "code",
-   "execution_count": 3,
-   "metadata": {
-    "collapsed": false
-   },
-   "outputs": [
-    {
-     "name": "stdout",
-     "output_type": "stream",
-     "text": [
-      "Parameter search took 53.03 seconds for 20 candidate parameter settings.\n",
-      "Model with rank: 1\n",
-      "Mean validation score: 0.552 (std: 0.006)\n",
-      "Parameters: {'l': 0.01, 'l2': 0.006918202991034834, 'power_t': 0.359507900573786}\n",
-      "\n",
-      "Model with rank: 2\n",
-      "Mean validation score: 0.541 (std: 0.012)\n",
-      "Parameters: {'l': 0.1, 'l2': 0.006767667154456677, 'power_t': 0.6706378696181594}\n",
-      "\n",
-      "Model with rank: 3\n",
-      "Mean validation score: 0.541 (std: 0.009)\n",
-      "Parameters: {'l': 0.01, 'l2': 0.005318483217500717, 'power_t': 0.4146619399905236}\n",
-      "\n"
-     ]
-    }
-   ],
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
    "source": [
     "# Parameter Grid Search\n",
-    "# http://scikit-learn.org/stable/auto_examples/model_selection/randomized_search.html#example-model-selection-randomized-search-py\n",
+    "# https://scikit-learn.org/stable/auto_examples/model_selection/plot_randomized_search.html#sphx-glr-auto-examples-model-selection-plot-randomized-search-py\n",
     "\n",
+    "import pandas as pd\n",
     "from operator import itemgetter\n",
     "from time import time\n",
-    "from sklearn.grid_search import RandomizedSearchCV\n",
+    "from sklearn.model_selection import RandomizedSearchCV\n",
```

Comment:
is there anywhere to specify the min version of sklearn supported? Not sure if this file is outdated https://github.com/VowpalWabbit/vowpal_wabbit/blob/5cba65d53d514a9f6273d12d059f524829393803/requirements.txt#L3

Reply:
Certainly the file seems outdated; a good fix would be to update it to `>=0.22.0` which would imply that we need at least the version  0.22.0 for smooth working.

A small recommendation would be to refrain from changing the requirements.txt in this PR since I think we should keep minimum requirement updates in a separate PR in case something breaks :)

Reply:
As per discussion with @peterychang , this is not to be updated at the moment.

Reply:
If you want to make this compatible with both the older and newer versions of sklearn, you can conditionally import the object based on the version of sklearn being used

Reply:
What do you say would be a good way to do the same; a try-catch block or checking the version and then making the import? I prefer the former.

Reply:
I've used the former method in 4befc30 to implement the same.